docs/sources/flow/setup/deploy-agent.md

```diff
@@ -0,0 +1,125 @@
+---
+
+# Deploying Grafana Agent
+
+Grafana Agent is a flexible, vendor-neutral telemetry collector. This
+flexibility means that the Agent doesn’t enforce a specific deployment topology
+but can work in multiple scenarios.
+
+This page lists common topologies used for deployments of Grafana Agent, when
+to consider using each, issues that users may run into, as well as scaling
+considerations.
+
+## As a centralized collection service
+Deploying the Grafana Agent as a centralized service is recommended for
+collecting application telemetry. This allows a smaller set of agents to
+coordinate service discovery, collection, and remote writing.
+
+![](../../../assets/deploy-agent/centralized-collection.png)
+
+Using this topology requires deploying the Agent on separate infrastructure,
+and making sure that agents can discover and reach these applications over the
+network. The main predictor for the size of the agent is the number of active
+metrics series it is scraping; a rule of thumb is ~10kb of memory for each
+series. We recommend you start looking towards horizontal scaling around the 1M
+active series mark.
+
+### Using Kubernetes StatefulSets
+Deploying Grafana Agent as a StatefulSet is the recommended option for metrics
+collection.
+The persistent pod identifiers make it possible to consistently match volumes
+with pods so that you can use them for the WAL directory.
+
+You can also use a Kubernetes deployment in cases where persistent storage is not required, such as a traces-only pipeline.
+
+### Pros
+* Straightforward scaling using [clustering][] or [hashmod sharding][]
+* Minimizes the “noisy neighbor” effect
+* Easy to meta-monitor
+
+### Cons
+* Requires running on separate infrastructure
+
+### Use for
+* Scalable telemetry collection
+
+### Don’t use for
+* Host-level metrics and logs
+
+## As a host daemon
+Deploying one Grafana Agent per machine is required for collecting
+machine-level metrics and logs, such as node_exporter hardware and network
+metrics or journald system logs.
+
+![](../../../assets/deploy-agent/daemonset.png)
+
+Each Agent requires opening an outgoing connection for each remote endpoint
+it’s shipping data to. This can lead to NAT port exhaustion on the egress
+infrastructure. Each egress IP can support up to (65535 - 1024 = 64511)
+outgoing connections on different ports. So, if all agents are shipping metrics
+and log data, an egress IP can support up to 32,255 agents.
+
+### Using Kubernetes DaemonSets
+The simplest use case of the host daemon topology is a Kubernetes DaemonSet,
+and is required for node-level observability (for example cAdvisor metrics) and
+collecting pod logs.
+
+### Pros
+* Doesn’t require running on separate infrastructure
+* Typically leads to smaller-sized agents
+* Lower network latency to instrumented applications
+
+### Cons
+* Requires planning a process for provisioning the Agent on new machines, as well as keeping configuration up to date to avoid configuration drift
```

Comment:
```suggestion
* Requires planning a process for provisioning Grafana Agent on new machines, as well as keeping configuration up to date to avoid configuration drift
```